packages/rum/src/apm-base.js

```diff
@@ -99,9 +99,33 @@ class ApmBase {
     return !this._disable
   }
 
+  /**
+   * When the required config keys are invalid, the agent is deactivated with
+   * logging error to the console
+   */
   config(config) {
-    var configService = this.serviceFactory.getService('ConfigService')
+    const configService = this.serviceFactory.getService('ConfigService')
     configService.setConfig(config)
+    /**
+     * Skip validation when agent is not active
+     */
+    if (!configService.isActive()) {
```

Comment:
This is when agent is inactive, We should not do any extra work when 
```
apm.init({active: false}) 
```

is on the page, It should basically be a noop.

Reply:
Validation should run whenever the agent is started and config is set. Users would be using our API and wont be calling set config manually. Plus this aligns with other agents like `Node.js`. Only thing missing is deactivating the agent when config is invalid which I added as TODO and will go in next breaking change.

Reply:
What I'm worried about is that we will end up in an invalid state if the agent is activated again. IMO we should either not set the config at all or set it with validation. I don't consider this too much of an extra work after all the user is calling this function and expecting to set the config.

Reply:
We validate the agent and log errors when the agent is activated again. Yeah its not a lot of work, The reason why I skipped the validation is to stop the errors from being getting logged since the agent is anyways inactive.